Gems/ROS2/Code/Source/Clock/SimulationClock.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <builtin_interfaces/msg/time.hpp>
+#include <rclcpp/publisher.hpp>
+#include <rosgraph_msgs/msg/clock.hpp>
+
+namespace ROS2
+{
+    //! Simulation clock which can tick and serve time stamps.
+    class SimulationClock
+    {
+    public:
+        //! Get simulation time as ROS2 message.
+        //! @see ROS2Requests::GetROSTimestamp() for more details.
+        builtin_interfaces::msg::Time GetROSTimestamp() const;
+
+        // TODO - consider having it called internally, also perhaps in a thread with a given frequency
```

Comment:
This TODO should be made into a GHI that can be addressed at a later point per the coding standard.